tutorials/juliacon2022.jl

```diff
@@ -0,0 +1,1644 @@
+the rank and name of the last element...
+"""
+
+# ╔═╡ 239406a6-0424-4107-a809-2fccb1dfb8b1
+lacto_ranks = split(comm_df.taxon[4], '|')
+
+# ╔═╡ 1fb80604-2385-49d6-b07c-b6b9f467c616
+# get the components of the last element
+lacto_rank, lacto_name = split(last(lacto_ranks), "__")
+
+# ╔═╡ 5456d0d6-2b69-4a91-9ef2-0ed55361c39f
+Taxon(lacto_name, :order)
+
+# ╔═╡ bde7469f-4734-4882-bd3b-2971468dfa3e
+md"""
+... or we can use the `taxon()` function,
+which understands this commonly used format and will build it for us:
+"""
+
+# ╔═╡ a42a53d3-8046-4c2a-bbef-4cd4d246b141
+taxon(last(lacto_ranks))
+
+# ╔═╡ 9812b45d-4973-414e-a71c-22dd81367f88
+md"""
+So, to get a vector of `Taxon`s, we can do the following:
+"""
+
+# ╔═╡ 064c6b27-f16d-4fbd-9ec6-67e091ecc9d2
+taxa = map(comm_df.taxon) do t
+	ranks = split(t, '|')
+	taxon(last(ranks))
+end
+
+# ╔═╡ 4e282e00-0749-4c0a-b44d-ef3cdfbdb10f
+md"""
+### The `MicrobiomeSample` type
+
+Biosamples in `Microbiome.jl` index the columns of our abundance matrix,
+and can also store optional metadata.
+Let's get the sample names from our file names,
+and then add the body site as metadata.
+
+We'll use the first sample as an example:
+"""
+
+# ╔═╡ 83ebfa19-b6d6-40db-9cae-bb5e840b4e46
+stool_file = first(files)
+
+# ╔═╡ 36591c12-f038-4ea7-af94-1e60b7aad0ba
+(samplename, site, ext) = split(stool_file, r"(-|_profile)"; limit = 3)
+
+# ╔═╡ 0af2ae11-cc91-44a2-8fb9-cef16d15fbee
+stool_samp = MicrobiomeSample(samplename)
+
+# ╔═╡ df70cfe0-2859-4b22-ba01-ed12db675e27
+md"""
+Sample metadata is stored in a `Dictionary` from the `Dictionaries.jl` package.
+We can add metadata using `set!` or `insert!`
+(the latter will throw an error if the metadata already exists):
+"""
+
+# ╔═╡ 4de0df7b-457d-4a38-a47f-83e536c47473
+set!(stool_samp, :body_site, site)
+
+# ╔═╡ 8715a036-c3b3-49bd-af31-d349c0d63bc3
+md"""
+So let's do this for all of our samples:
+"""
+
+# ╔═╡ b9d102ac-0aab-45f6-9f1d-f23ba590bcc6
+mbiom_samples = map(files) do f
+	(name, site, ext) = split(f, r"(-|_profile)"; limit = 3)
+	s = MicrobiomeSample(name)
+	insert!(s, :body_site, site)
+	s
+end
+
+# ╔═╡ 5e77d309-ca1b-47b2-8c71-a0cb2c96758a
+md"""
+### Constructing a `CommunityProfile`
+
+Now have all the elements, and we just need to put them together:
+"""
+
+# ╔═╡ 96ca485b-db0a-4517-b15b-6cb3232c86e2
+comm = CommunityProfile(mat, taxa, mbiom_samples)
+
+# ╔═╡ f5decd43-03ca-40ce-89a0-2cb74fa0fa70
+md"""
+## Working with `CommunityProfile`s
+
+Now that we have our `CommunityProfile`, there are a number of ways to work with it.
+First, let's take a look at the various accessor functions.
+
+- `features()` - returns a vector of the features (in this case `Taxon`s)
+- `samples()` - returns a vector of the samples, including the attached metadata
+- `featurenames()`/`samplenames()` - like `features()` and `samples()`, but returns just the string representation.
+- `abundances()` - returns the sparse matrix with relative abundances`
```

Comment:
should there be an example of using `abundances()` below?

Reply:
Nah, it's straightforward